src/components/Verify.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Redirect } from 'react-router';
+import Shell from './shared/Shell';
+import Card from './shared/Card';
+import { SyncLoader } from 'react-spinners';
+import { Mutation } from 'react-apollo';
+import { VERIFY } from '../graphql/mutations';
+import config from '../config';
+import qs from 'query-string';
+
+const EVENT_COLOR = config.EVENT_MAIN_COLOR;
+
+/* This class exists so that the mutation can take place when the component mounts
+   as per this issue: https://stackoverflow.com/questions/49456738/how-to-run-a-mutation-on-mount-with-react-apollo-2-1s-mutation-component
+*/
+class CallVerify extends React.Component {
```

Comment:
Is this extra class needed? Could you not add this piece of code in the Verify class component?

Reply:
The reason I create another class is due to the fact that I need to perform the function once a component is mounted. Before attempting to use another Class component and just rendering through a function, there would be state changes occurring improperly throwing me an error. After searching through fixes, a common fix is to perform the function through another class. If you have any other ideas let me know.